build/stf-run-ci/tasks/setup_base.yml

```diff
@@ -26,20 +26,43 @@
         targetNamespaces:
         - "{{ namespace }}"
 
-- name: Subscribe to AMQ Certificate Manager Operator
-  k8s:
-    definition:
-      apiVersion: operators.coreos.com/v1alpha1
-      kind: Subscription
-      metadata:
-        name: amq7-cert-manager-operator
-        namespace: openshift-operators
-      spec:
-        channel: 1.x
-        installPlanApproval: Automatic
-        name: amq7-cert-manager-operator
-        source: redhat-operators
-        sourceNamespace: openshift-marketplace
+- block:
+  - name: Create openshift-cert-manager-operator namespace
+    k8s:
+      definition:
+        apiVersion: project.openshift.io/v1
+        kind: Project
+        metadata:
+          name: openshift-cert-manager-operator
+        spec:
+          finalizers:
+          - kubernetes
+
+  - name: Create openshift-cert-manager-operator OperatorGroup
+    k8s:
+      definition:
+        apiVersion: operators.coreos.com/v1
+        kind: OperatorGroup
+        metadata:
+          generateName: openshift-cert-manager-operator-
+          name: openshift-cert-manager-operator
```

Comment:
I think you want either generateName OR name, not both.

https://docs.openshift.com/container-platform/4.10/rest_api/objects/index.html#io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta says "GenerateName is an optional prefix, used by the server, to generate a unique name ONLY IF the Name field has not been provided"

I tested this and found that the resource as shown is always called openshift-cert-manager-operator (generateName is ignored). If I remove the name field, a unique one is generated:

```
(ansible) [csibbitt@csibbitt rhos-qe-jenkins]$ oc get operatorgroup
NAME                                    AGE
openshift-cert-manager-operator         57s
openshift-cert-manager-operator-bql8b   4s
```
I'm not sure which you wanted (just the name is probably fine, I don't think we want multiple operatorGroups), but having both doesn't make sense.

Reply:
Yea what I wanted was the operatorgroup to be unique (not that it matters in our test rig) as that's how the OLM deployment would end up creating it. I was following this example from the official docs:

https://docs.openshift.com/container-platform/4.9/hardware_enablement/psap-special-resource-operator.html

Unfortunately I started with generateName only, and it doesn't work on a fresh deployment when the OperatorGroup doesn't already exist apparently (https://github.com/openshift/openshift-restclient-python/issues/279). So I added the name and it worked.

You're right though, a dynamically generated OperatorGroup name is unnecessary and not something we do in other parts of the playbook, so removal of the superfluous `generateName` is appropriate. I'll make the change since it's small and might as well be consistent.

I suspect it worked for you because you created the manifest directly using `oc create` via the `oc` CLI client? (or maybe because you already had the other operatorgroup?)

Reply:
Yeah it was done with `oc create` so not affected by that openshift-restclient-python issue perhaps.

Just for science, I checked and it doesn't make a difference whether there is an existing one or not.

```
$ oc create -f -
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
    generateName: openshift-cert-manager-operator-
    namespace: default
spec: {}
operatorgroup.operators.coreos.com/openshift-cert-manager-operator-7dqbp created

$ oc get operatorgroup
NAME                                    AGE
openshift-cert-manager-operator-7dqbp   11s
$ 
```